benchmark_runner/jupyterlab/templates/summary_report/summary_report_operations.py

```diff
@@ -0,0 +1,353 @@
+
+            geometric_mean_df_latency = pd.DataFrame(self.geometric_mean_data_latency)
+            self.median_indices = geometric_mean_df_latency.groupby(['metric', 'ocp_version']).apply(self.__get_median)
+            geometric_mean_df_latency = geometric_mean_df_latency.loc[self.median_indices['median_result']]
+            geometric_mean_df_latency = self.__calc_percentage(geometric_mean_df_latency, complementary=True)
+
+            # Concatenate DataFrames along rows (assuming the keys are the same)
+            self.geometric_mean_df = pd.concat([geometric_mean_df_throughput, geometric_mean_df_latency], ignore_index=True)
+        elif workload == 'vdbench':
+            geometric_mean_df_iops = pd.DataFrame(self.geometric_mean_data_iops)
+            self.median_indices = geometric_mean_df_iops.groupby(['metric', 'ocp_version']).apply(self.__get_median)
+            geometric_mean_df_iops = geometric_mean_df_iops.loc[self.median_indices['median_result']]
+            geometric_mean_df_iops = self.__calc_percentage(geometric_mean_df_iops)
+
+            geometric_mean_df_latency = pd.DataFrame(self.geometric_mean_data_latency)
+            self.median_indices = geometric_mean_df_latency.groupby(['metric', 'ocp_version']).apply(self.__get_median)
+            geometric_mean_df_latency = geometric_mean_df_latency.loc[self.median_indices['median_result']]
+            self.update_operator_versions(geometric_mean_df_latency)
+            geometric_mean_df_latency = self.__calc_percentage(geometric_mean_df_latency, complementary=True)
+
+            # Concatenate DataFrames along rows (assuming the keys are the same)
+            self.geometric_mean_df = pd.concat([geometric_mean_df_iops, geometric_mean_df_latency], ignore_index=True)
+        elif workload == 'bootstorm' or workload == 'windows':
+            self.geometric_mean_df = pd.DataFrame(self.geometric_mean_data)
+            self.median_indices = self.geometric_mean_df.groupby(['metric', 'ocp_version']).apply(self.__get_median)
+            self.geometric_mean_df = self.geometric_mean_df.loc[self.median_indices['median_result']]
+            self.geometric_mean_df = self.__calc_percentage(self.geometric_mean_df, complementary=True)
+            # updates metric names
+            if workload == 'windows':
+                windows_os = ['windows10', 'windows11', 'windows_server_2019', 'windows_server_2022']
+                for os in windows_os:
+                    self.geometric_mean_df['metric'] = self.geometric_mean_df['metric'].replace({os: f'p90 ({os})'})
+            else:
+                self.geometric_mean_df['metric'] = self.geometric_mean_df['metric'].replace(
+                    {'fedora37': 'p90 (fedora37)'})
+
+    @typechecked()
+    def get_median_geometric_mean_df(self, workload: str, df: pd.DataFrame):
+        """
+        This method calculates the geometric mean per workload
+        @param workload:
+        @param df:
+        @return:
```

Comment:
I don't understand what this means.  What is the "median geometric mean"?  What does "df" mean?

Reply:
median geometric mean: This method calculates the geometric mean per workload from input dataframe , I updated it in the comment.

Reply:
I dont see any reason to move it to separate methods, we just prepare the workload dataframe template and its different between each workload.